rma/models/rma.py

```diff
@@ -317,6 +317,107 @@ def _domain_location_id(self):
         copy=False,
     )
 
+    show_create_receipt = fields.Boolean(
+        string="Show Create Receipt Button", compute="_compute_show_create_receipt"
+    )
+    show_create_return = fields.Boolean(
+        string="Show Create Return Button", compute="_compute_show_create_return"
+    )
+    show_create_replace = fields.Boolean(
+        string="Show Create replace Button", compute="_compute_show_create_replace"
+    )
+    show_create_refund = fields.Boolean(
+        string="Show Create refund Button", compute="_compute_show_refund_replace"
+    )
+
+    # Replace fields, used when the delivery is created automatically
+    replace_warehouse_id = fields.Many2one(
```

Comment:
The computation of these fields will mean an overhead on existing installations. Why they are needed? This seems a new feature that is not linked with the configurable workflow.

Reply:
In the operation, if the user selects 'Automatic Delivery' and 'Different Return Product,' the workflow will trigger the replacement action. Since the replacement action requires user input, the replacement information should be filled in the RMA at the time of its creation.

The computed fields for the replacement warehouse and replacement product UoM are intended to provide default values. A migration script can be added to create these columns, ensuring that their computation does not trigger on existing RMAs.

![image](https://github.com/user-attachments/assets/d8a1f98f-fd15-4ce8-9b59-18e8b187aaa6)

Reply:
Please add the pot_init_hook/migration script thus.

Reply:
done

Reply:
@pedrobaeza ,

@jbaudoux  pointed out that replacing a product with a different one can be handled by processing an RMA reception with a refund and then creating a normal sale. I will remove this part from the current workflow.

Another use case reported by the customer is receiving a different product than the one ordered. In this case, we need to receive a product that is not listed on the sale order and then redeliver the originally ordered product.

I am considering adding a new field, 'product to return,' which will be used if the operation has 'return a different product' set to True.
In this case, the reception will involve the different product. Otherwise, the standard behavior will apply, and the product_id will be used for both reception and delivery.

What are your thoughts on this?